avogadro/mainwindow.cpp

```diff
@@ -1340,15 +1342,24 @@ bool MainWindow::saveFileAs(bool async)
   dir = info.absoluteDir().absolutePath();
   settings.setValue("MainWindow/lastSaveDir", dir);
 
-  // Create one of our writers to save the file:
+  // Use manually entered extension if present
   QString extension = info.suffix().toLower();
+  // Otherwise, get extension from selected filter
+  if (extension.isEmpty()) {
+    QString filter = saveDialog.selectedNameFilter();
+    if (filter.contains("(*.cjson)", Qt::CaseSensitive))
+      extension = "cjson";
+    else if (filter.contains("(*.cml)", Qt::CaseSensitive))
+      extension = "cml";
+    fileName += "." + extension;
```

Comment:
Right, but if we don't select something in the filter *and* don't add an extension, we should default to adding ".cjson"